### ISSUE TITLE
Design Question?

### ISSUE THREAD

**Issue by Outstep** [ipfs-search/ipfs-search#165]
Hello,

I am learning about IPFS and now also your ipfs-search engine but am not clear as to where the indexes are actually being stored.

In my reading on your engine, I seems that you are indexing the IPFS network, but that the indexes are being stored locally in an ElasticSearch database (elasticsearch-oss:7.8.1) from what I read in the docker-compose.yml file.

Forgive the naivety here, but I thought that part of the effort was to also store the indices across the IPFS network as well.

Did I miss somethin here?
Thanks

**Comment by dokterbob** [comment]
Thanks for your question and your interest in our project. We are in fact publishing snapshots of our index, in Elasticsearch format, to IPFS. See our documentation here: https://github.com/ipfs-search/ipfs-search/blob/master/docs/snapshots.md

However, actually decentralising search is very hard. Our database is over 1 TB now and is expected to scale as IPFS does. Hence, soon it will not be possible for any single computer to host the index, at which point it will require a cluster. 

We do hope to start experimenting with decentralisation of at least our index, but this is a truly difficult challenge. Specifically, after searching through the data (which can be decentralised fairly efficiently), the results need to be joined and sorted. This needs to be done at a central place and often involves a very large number of results. If this place is your computer, it would mean that you'd have to request content from several hundred nodes and it would receive a large amount of data, it would then need to process it and hold it in memory just to show a single page.

Of course there are shortcuts; not searching through the full content, not bothering with ranking, making crude estimates etc. but this sort of tuning is extremely complicated, even for a 'centralised' system like Elasticsearch clusters, with sub-millisecond delays between nodes. 

In short; we are working on it, aware of the problem but choose to have a working search engine right now while giving the 'right to fork' by sharing our dataset.
